pyproject.toml

```diff
@@ -66,7 +66,8 @@ Homepage = "https://github.com/scikit-hep/pyhf"
 [project.optional-dependencies]
 shellcomplete = ["click_completion"]
 tensorflow = [
-    "tensorflow>=2.7.0",  # c.f. PR #1962
+    "tensorflow>=2.7.0; platform_machine != 'arm64'",  # c.f. PR #1962
+    "tensorflow-macos>=2.7.0; platform_machine == 'arm64' and platform_system == 'Darwin'",  # c.f. PR #1962
```

Comment:
Purposefully over specified with both `platform_machine` and `platform_system` as I do not know for certain that `arm64` will be reserved for Apple silicon machines forever.